src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiVariableConfigurationEditor.js

```diff
@@ -0,0 +1,214 @@
+import {FieldArray, useFormikContext} from 'formik';
+import isEqual from 'lodash/isEqual';
+import PropTypes from 'prop-types';
+import React, {useContext} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useAsync, useToggle} from 'react-use';
+
+import {APIContext} from 'components/admin/form_design/Context';
+import {REGISTRATION_OBJECTS_TARGET_PATHS} from 'components/admin/form_design/constants';
+import Field from 'components/admin/forms/Field';
+import Fieldset from 'components/admin/forms/Fieldset';
+import FormRow from 'components/admin/forms/FormRow';
+import {TextInput} from 'components/admin/forms/Inputs';
+import Select, {LOADING_OPTION} from 'components/admin/forms/Select';
+import {post} from 'utils/fetch';
+
+import {asJsonSchema} from './utils';
+
+/**
+ * Returns the Objects API Configuration editor modal for a specific variable. This only applies to V2 Options
+ *
+ * @typedef {{
+ *   version: 1 | 2;
+ *   objecttype: string;
+ *   objecttypeVersion: number;
+ *   variablesMapping: {variableKey: string, targetPath: string[]}[];
+ * }} ObjectsAPIRegistrationBackendOptions
+ *
+ * @param {Object} p
+ * @param {Object} p.variable - The current variable
+ * @returns {JSX.Element} - The summary, represented as a the parts of the target path separated by '>'
+ */
+const ObjectsApiVariableConfigurationEditor = ({variable}) => {
+  const {csrftoken} = useContext(APIContext);
+
+  const [jsonSchemaVisible, toggleJsonSchemaVisible] = useToggle(false);
+  const {values: backendOptions, getFieldProps, setFieldValue} = useFormikContext();
+
+  /** @type {ObjectsAPIRegistrationBackendOptions} */
+  const {objecttype, objecttypeVersion, variablesMapping, version} = backendOptions;
+  if (version !== 2) throw new Error('Not supported, must be config version 2.');
+
+  // get the index of our variable in the mapping, if it exists
+  let index = variablesMapping.findIndex(
+    mappedVariable => mappedVariable.variableKey === variable.key
+  );
+
+  if (index === -1) {
+    // if not found, grab the next available index to add it as a new record/entry
+    index = variablesMapping.length;
+  }
+
+  let mappedVariable = variablesMapping[index] || {
+    variableKey: variable.key,
+    targetPath: undefined,
+  };
+  // the formik state is populated with the backend options, so our path needs to be
+  // relative to that
+  const namePrefix = `variablesMapping.${index}`;
+
+  const {
+    loading,
+    value: targetPaths,
+    error,
+  } = useAsync(
+    async () => {
+      const response = await post(REGISTRATION_OBJECTS_TARGET_PATHS, csrftoken, {
+        objecttypeUrl: objecttype,
+        objecttypeVersion,
+        variableJsonSchema: asJsonSchema(variable),
+      });
```

Comment:
I think that the post already does the `throwForStatus` thing, so I am not sure if the if statement below is needed?

Reply:
I will test when I add proper error handling in a follow up PR

Reply:
Wasn't necessary